common4j/src/main/com/microsoft/identity/common/java/util/StringUtil.java

```diff
@@ -45,6 +60,138 @@ public static boolean isNullOrEmpty(String message) {
         return message == null || message.trim().length() == 0;
     }
 
+    /**
+     * Returns true if the string contains the given substring.
+     *
+     * @param str    the string to search for a substring in
+     * @param substr the substring to search for
+     * @return true if the string contains the given substring, false if it does not or if it is null
+     */
+    public static boolean containsSubString(final String str, final String substr) {
+        if (isNullOrEmpty(str)) {
+            return false;
+        }
+
+        return str.contains(substr);
+    }
+
+    /**
+     * Utility to null-safe-compare string in a case-insensitive manner.
+     *
+     * @param one The first string to compare.
+     * @param two The second string to compare.
+     */
+    public static boolean equalsIgnoreCase(@Nullable final String one, @Nullable final String two) {
+        return Objects.equals(one, two) || (one != null && one.equalsIgnoreCase(two));
+    }
+
+    /**
+     * Parses the supplied home_account_id to extract the uid, utid.
+     *
+     * @param homeAccountId The home_account_id to inspect.
+     * @return A Pair of Strings representing the uid/utid of the supplied home_account_id.
+     * Return value cannot be null, but its values (pair.first, pair.second) may be.
+     */
+    public static Map.Entry<String, String> getTenantInfo(@NonNull final String homeAccountId) {
+        // Split this value by its parts... <uid>.<utid>
+        final int EXPECTED_LENGTH = 2;
+        final int INDEX_UID = 0;
+        final int INDEX_UTID = 1;
+
+        final String[] uidUtidArray = homeAccountId.split("\\.");
+
+        String uid = null;
+        String utid = null;
+
+        if (EXPECTED_LENGTH == uidUtidArray.length
+                && !isNullOrEmpty(uidUtidArray[INDEX_UID])
+                && !isNullOrEmpty(uidUtidArray[INDEX_UTID])) {
+            uid = uidUtidArray[INDEX_UID];
+            utid = uidUtidArray[INDEX_UTID];
+        }
+
+        return new AbstractMap.SimpleEntry<>(uid, utid);
```

Comment:
Ok, there's no pair being returned here...and I'm not sure I like it being a `Map.Entry` either..I mean I believe Map.Entry is meant to be a key and value...not sure if `uid` is a valid key for `utid`....isntead of doing this..can we just create a dedicated class for TenantInfo?

```java
class TenantInfo {
   private final String uid;
   private final String utid;
}
```

Reply:
Many of the code are taken/migrated as-is, and i'm refraining myself from modifying them (unless i think it really should be improved right away) - otherwise the scope of change will keep growing (and more test will be required).

LMK if i definitely should pick this up.

Reply:
Let's leave it as is for now then